procurement/tests/test_email_alerts.py

```diff
@@ -0,0 +1,209 @@
+            uuid="ABC3",
+            title="Test Tender 3",
+            description="This is a test tender!",
+            state="This is a test state",
+            value=10000,
+            published=now-dt.timedelta(days=5),
+            start_date=dt.datetime(year=2022, month=12, day=1),
+            end_date=dt.datetime(year=2022, month=12, day=31),
+            council=Council.objects.get(name="Ceredigion County Council")
+        )
+
+        Tender.objects.create(
+            uuid="ABC4",
+            title="Test Tender 4",
+            description="This is a test tender!",
+            state="This is a test state",
+            value=10000,
+            published=now-dt.timedelta(days=12),
+            start_date=dt.datetime(year=2022, month=12, day=1),
+            end_date=dt.datetime(year=2022, month=12, day=31),
+            council=Council.objects.get(name="Aberdeenshire Council")
+        )
+
+        Tender.objects.create(
+            uuid="ABC5",
+            title="Test Tender 5",
+            description="This is a test tender!",
+            state="This is a test state",
+            value=10000,
+            published=now-dt.timedelta(days=14),
+            start_date=dt.datetime(year=2022, month=12, day=1),
+            end_date=dt.datetime(year=2022, month=12, day=31),
+            council=Council.objects.get(name="Bristol City Council")
+        )
+
+        Classification.objects.create(
+            description="Test classification!",
+            classification_scheme="Test scheme!",
+            group="A"
+        )
+
+        Classification.objects.create(
+            description="Test classification!",
+            classification_scheme="Test scheme!",
+            group="B"
+        )
+
+        Classification.objects.create(
+            description="Test classification!",
+            classification_scheme="Test scheme!",
+            group="C"
+        )
+
+        TenderClassification.objects.create(
+            tender=Tender.objects.get(uuid="ABC5"),
+            classification=Classification.objects.get(group="C")
+        )
+
+        TenderClassification.objects.create(
+            tender=Tender.objects.get(uuid="ABC4"),
+            classification=Classification.objects.get(group="C")
+        )
+
+        TenderClassification.objects.create(
+            tender=Tender.objects.get(uuid="ABC3"),
+            classification=Classification.objects.get(group="B")
+        )
+
+        TenderClassification.objects.create(
+            tender=Tender.objects.get(uuid="ABC2"),
+            classification=Classification.objects.get(group="A")
+        )
+
+        TenderClassification.objects.create(
+            tender=Tender.objects.get(uuid="ABC1"),
+            classification=Classification.objects.get(group="B")
+        )
+
+
+    def test_regions_filter_with_country(self):
+        qs = Tender.objects.all()
+        f = TenderFilter(data={"region": "England"}, queryset=qs)
```

Comment:
oooh, nice. It had never occurred to me to test filters directly like this.

Reply:
It tests the backend effectively, but it doesn't test the UI particularly. I'm not sure how best to go about this with any sort of automated testing.